corehq/apps/smsbillables/models.py

```diff
@@ -367,3 +399,51 @@ def _get_usage_fee(cls, domain, direction):
                 % (direction, domain)
             )
         return usage_fee
+
+    @classmethod
+    def _get_twilio_charges(cls, backend_message_id, direction, couch_id):
+        def _get_twilio_client():
+            twilio_backend = SQLTwilioBackend.get_global_backends_for_this_class(SQLTwilioBackend.SMS).get()
+            config = twilio_backend.config
+            return TwilioRestClient(config.account_sid, config.auth_token)
+
+        try:
+            if backend_message_id:
+                twilio_message = _get_twilio_client().messages.get(backend_message_id)
```

Comment:
i think what we can do is raise an exception, something like `RetryBillableTask`, and then catch it in the celery task definition where you can tell it to retry the task a minute later. for an example of creating a retryable task with celery, see https://github.com/dimagi/commcare-hq/blob/e2dd3c78a6906a6b82f749e9cee9c64fd7581d2a/corehq/apps/sms/tasks.py#L271. we can tell it how many times to retry and how long to wait before retrying.